lib/KCard/index.vue

```diff
@@ -0,0 +1,314 @@
+<template>
+
+  <BaseCard
+    :to="to"
+    :title="title"
+    :headingLevel="headingLevel"
+    :titleLines="titleLines"
+    :style="wrapperStyle"
+  >
+    <template #title>
+      <!-- @slot Optional slot section containing the title contents, should not contain a heading element. -->
+      <div class="title-slot">
+      <!-- <slot name="title"></slot> -->
+      </div>
+    </template>
+
+    <template #default>
+      <aside
+        v-if="thumbnailDisplay !== 'none'"
+        :style="{ dynamicOrder }"
+        :class="{
+          'image-class-vertical': layout === 'vertical',
+          'image-class-horizontal': (layout === 'horizontal' && thumbnailDisplay === 'small')
+        }"
+      >
+        <KImg
+          v-if="thumbnailSrc"
+          :src="thumbnailSrc"
+          :isDecorative="true"
+          :appearanceOverrides="{ scaleType: thumbnailScaleType, width: cardContentPartial }"
+          class="thumbnail-image"
+          :style="imageRadius"
+        />
+        <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+      </aside>
+      <div class="display-inline-block" :style="cardContentPartial">
+        <div
+          data-testid="aboveTitle"
+          class="above-title-style"
+          :class="{ 'above-title-horizontal': layout === 'horizontal' }"
+        >
+          <slot name="aboveTitle"></slot>
```

Comment:
@MisRob I wonder if these "aboveTitle", "belowTitle" slots should also be present on the  BaseCard component? I dont see these slots in the BaseCard specs. But we dont have a way of control these slots here in the KCard to make them appear above or below the title, since it is the BaseCard who has the responsibility of rendering it, so probably its also the BaseCard the responsilble for rendering it.

Reply:
@AlexVelezLl There's a way to control them via styling with flex order which is needed for more reasons anyways - primarily because the placement of slots in the code really shouldn't change the resulting markup structure. However, even though it's technically valid and possible, it creates some complexity when `KCard` needs to do some styling in the `BaseCard`, so I hear you. If the separation of `KCard` and `BaseCard` brings more cons than pros, an alternative that would resolve all this would be to merge these two components into a single `KCard` component. I'm open to this and leave to @AllanOXDi's judgement.

Reply:
Note that even if we merged those two components though, we'd still need to show the content of `aboveTitle` below the title in the markup and achieve placing it above the title by means of CSS rather than HTML. This is a standard a11y friendly coding technique and a core feature of the card ensuring that screen readers announce it as expected.

So the question of merging vs not would really be about cross-component vs one-component styling. In both cases we'll still be using `flex-order`.